sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RequestIdPolicy.java

```diff
@@ -3,56 +3,46 @@
 
 package com.azure.core.http.policy;
 
-import com.azure.core.http.HttpHeader;
 import com.azure.core.http.HttpPipelineCallContext;
 import com.azure.core.http.HttpPipelineNextPolicy;
 import com.azure.core.http.HttpResponse;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpHeader;
 import reactor.core.publisher.Mono;
 
 import java.util.Objects;
 import java.util.UUID;
-import java.util.function.Supplier;
 
 /**
- * The pipeline policy that puts a UUID in the request header. Azure uses the request id as
- * the unique identifier for the request.
+ * The pipeline policy that puts a UUID in the request header. The default {@link HttpHeader} name can be overwritten.
+ * Azure uses the request id as the unique identifier for the request.
  */
 public class RequestIdPolicy implements HttpPipelinePolicy {
-    private static final String REQUEST_ID_HEADER = "x-ms-client-request-id";
-    private final Supplier<HttpHeader> requestIdSupplier;
 
+    private static final String REQUEST_ID_HEADER = "x-ms-client-request-id";
+    private final String requestIdHeaderName;
 
     /**
-     * Creates a RequestIdPolicy provided {@link Supplier}.
-     *
-     * @param headerSupplier The {@link Supplier} to provide header to add to outgoing requests.
+     * Creates  {@link RequestIdPolicy} with provided {@code requestIdHeaderName}.
+     * @param requestIdHeaderName to be used to set in {@link HttpRequest}.
      */
-    public RequestIdPolicy(Supplier<HttpHeader> headerSupplier) {
-        this.requestIdSupplier = Objects.requireNonNull(headerSupplier, "headerSupplier not be null.");
+    public RequestIdPolicy(String requestIdHeaderName) {
```

Comment:
The search team only need to override header name. This constructor will satisfy their ask.